### DISCUSSION TITLE
Install from source code

### DISCUSSION THREAD

**Discussion by liu781319869** [deepmodeling/deepmd-kit#3277] [General]
when I install deepmd-it in the source code. After the command "pip install .", an error is reported as shown in the figure. How to solve this problem? thank you!
![微信图片_20240215103340](https://github.com/deepmodeling/deepmd-kit/assets/52098480/04b3952e-c30c-4afd-a25c-eede82e86a7b)

**Comment by njzjz** [comment]
Looks like the same situation as https://github.com/pypa/pip/issues/11796#issuecomment-1433617778.

You may check if your Python version is compatible with typeguard>=4.

**Reply by liu781319869** [reply]
When I switched the python version to 3.8, I continued the same command with a different error than in 3.7, which was strange.Thank you!
![微信图片_20240216202820](https://github.com/deepmodeling/deepmd-kit/assets/52098480/caabbc59-f59a-4e7c-a7ee-858ca0bc67e8)
![微信图片_20240216202828](https://github.com/deepmodeling/deepmd-kit/assets/52098480/ac56af3c-e0b1-47a5-bdf4-14fa7d3223d2)
